code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -582,30 +582,84 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 	return 0
 
 /datum/reagents/proc/reaction(var/atom/A, var/method=TOUCH, var/volume_modifier=0, var/amount_override = 0, var/list/zone_sels = ALL_LIMBS)
-	for(var/datum/reagent/R in reagent_list)
-		var/amount_splashed = (R.volume+volume_modifier)
-		if (amount_override)
-			amount_splashed = amount_override
-		if(ismob(A) && R)
-			if(isanimal(A) && R)
+	if (isliving(A))
+		handle_reagents_mob_thermal_interaction(A, method, zone_sels, amount_override)
+	for (var/datum/reagent/R in reagent_list)
+		var/amount_splashed = amount_override ? amount_override : (R.volume + volume_modifier)
+		if (ismob(A))
+			if (isanimal(A))
 				R.reaction_animal(A, method, amount_splashed)
 			else
 				R.reaction_mob(A, method, amount_splashed, zone_sels)
-		if(isturf(A) && R)
+		else if (isturf(A))
 			R.reaction_turf(A, amount_splashed)
-		if(istype(A, /obj) && R)
+		else if (istype(A, /obj))
 			R.reaction_obj(A, amount_splashed)
 
 /datum/reagents/proc/reaction_dropper(var/atom/A, var/volume_modifier=0)
 
-	if(ismob(A))
+	if (ismob(A))
+		if (isliving(A))
+			handle_reagents_mob_thermal_interaction(A, TOUCH, TARGET_EYES)
 		for(var/datum/reagent/R in reagent_list)
 			R.reaction_dropper_mob(A)
 
-	if(istype(A, /obj))
+	else if(istype(A, /obj))
 		for(var/datum/reagent/R in reagent_list)
 			R.reaction_dropper_obj(A, R.volume+volume_modifier)
 
+#define SCALD_PAINFUL 15
+#define SCALD_AGONIZING 30
+
+/datum/reagents/proc/handle_reagents_mob_thermal_interaction(mob/living/L, method, list/zone_sels = ALL_LIMBS, volume_used)
+	if (!isliving(L))
+		return
+
+	var/ignore_thermal_prot = FALSE
+	if (method == INGEST) //Eating or drinking burns the mouth (head) regardless of targeting and isn't blocked by head thermal protection.
```

Comment:
Perhaps stomach damage too?

Reply:
As far as I know there's no stomach organ.
Diona being able to eat weeds is based on their liver I think.